### PR TITLE
Option to create a message keys file for autocomplete & type checking

### PR DESCRIPTION
**Pull request by davnicwil**
You can get autocomplete & type checking at point of usage.

This is pretty useful as it means you can statically verify that your won't get missing keys at runtime.

For example, using `react-intl`:

```
// suppose key "does_exist" is in keys.js and "does_not_exist" is not
import messageKeys from './lokalise/keys.js'
import { injectIntl } from 'react-intl'

type MessageKey = $Keys<typeof messageKeys> // enum pattern in flow

type ReactIntlMessageDescriptor = {|
  id: MessageKey, // here's the type check that forces you to use a key that exists
  description?: string,
  defaultMessage?: string,
|}

export type ReactIntlProps = {
  intl: {
    formatMessage: (
      descriptor: ReactIntlMessageDescriptor,
      values?: Object,
    ) => string,
  }
}

const Component = (props: ReactIntlProps) => (
  <div>
    <div>{props.intl.formatMessage({ id: 'does_exist' })}</div> // this is fine
    <div>{props.intl.formatMessage({ id: messageKeys.does_exist })}</div> // also fine, even better you get autocomplete from messageKeys
    <div>{props.intl.formatMessage({ id: 'does_not_exist' })}</div> // flow will complain
)
```

You might also be able to use this with other tooling to statically verify unused keys across a whole project, which would also be very useful.

### CODE REVIEW

File: src/main.js [hunk 1]
```diff
@@ -17,10 +19,15 @@ const main = async () => {
...
+    }
+
     console.log('Localization Updated')
   } catch (err) {
     console.log('Localization Update Failed', err)
   }
 }
 
-main()
+main()
```
Comment (carlossless) [inline src/main.js]: Missing newline.

File: README.md [hunk 2]
```diff
@@ -1,15 +1,34 @@
 # Lokalise - the unofficial node lokali.se client.
 
-This is all still very WIP.
```
Comment (carlossless) [inline README.md]: Although I am open to change this text, I would still like a disclaimer that this is still not recommended for production (although it's been battle tested for some time). Let me think of something else a bit later. For now let's leave this?

File: README.md [hunk 3]
```diff
@@ -1,15 +1,34 @@
...
-
-## How to use
+### Configuration
```
Comment (carlossless) [inline README.md]: I am a huge stickler for semantics, sometimes even at the cost of aesthetics. Let's leave this as a second level heading?

File: README.md [hunk 4]
```diff
@@ -1,15 +1,34 @@
...
+  type: 'json' | 'es6_module' | 'es5_module' (optional, default 'json') the type of the output keys file
+  flow: [boolean] (optional, default false) in the case {type} is a js module, adds a //@flow annotation at the top so that the keys can be used with flow type checking
 }
 ```
+
+Alternatively, passing `"keys_file": true` gives you the all default config - i.e. a `keys.json` file will be output
+
+### Run
+
+`yarn lokalise` or `npm run lokalise`
```
Comment (carlossless) [inline README.md]: Erm, there's no such "script" in the npm package :D. It's a executable, I don't think we need documentation on how to run it in the readme. Usage instructions should be generated by commander, we could add those once I brush them up.

File: README.md (hunk 2 above)
Comment (davnicwil) [reply README.md]: 👍

File: README.md (hunk 3 above)
Comment (davnicwil) [reply README.md]: 👍

File: README.md (hunk 4 above)
Comment (davnicwil) [reply README.md]: Ah ok - I didn't actually try `npm run` but `yarn` (which I am using) will fall through to running the binary for the package if there is no script present in packages.json, which is a pretty nice feature (basically you don't have to create one of those dummy `"package": "package"` scripts). I'll take out the npm one.

Comment (carlossless) [reply README.md]: Well, if you link it as a global executable via `npm install -g lokalise` or `yarn global install lokalise`, then you can use it wherever you like outside the context of package managers.

**Pull request merged**